server/services/prison/PrisonRegisterService.ts

```diff
@@ -13,12 +13,19 @@ export default class PrisonRegisterService {
   }
 
   async getActivePrisonsFromPrisonRegister(): Promise<Array<Prison>> {
+    let activePrisons: Array<PrisonDto>
     try {
-      const activePrisons = await this.prisonRegisterStore.getActivePrisons()
-      return activePrisons || this.retrieveAndCacheActivePrisons()
+      activePrisons = await this.getActivePrisonDtos()
     } catch (error) {
-      return this.retrieveAndCacheActivePrisons()
+      return error
```

Comment:
Possibly something to consider down the line - if we have already cached the prisons and this call is just to do the sporadic update, would we return the stale cached prisons rather than throw an error?

Reply:
No, this method is simply about getting the active prisons. The previous code would look to get them from redis and if that returned something a) they would be 'fresh' (ish) as its a 24 hour cache, and b) they would be returned to the caller
If redis returned nothing (null/undefined), then the previous code would call the `retrieveAndCacheActivePrisons` method which would get them from the API and cache them in redis ready for next time.
If either case threw an exception it would attempt to call `retrieveAndCacheActivePrisons` again (!!! - I dont know why I wrote it that way in the first place !!)

The new code still does all that, albeit in a new method `getActivePrisonDtos`. New method does as before (try redis first, else call `retrieveAndCacheActivePrisons`), and just throws if something goes wrong (it has no catch of its own)
The new catch block here returns the thrown error; and that seems to make all the existing tests happy (tests that test exactly this scenario of either redis or the API throwing an error)

Re: the 'sporadic update' we dont really have anything like that per se (and never did have)
I know we have an AC that says prisons will be refreshed after 24 hours - our implementation of that is basically this method. In that with an empty cache, it will get retrieved from prison API and cached with a TTL of 24 hours. Within those 24 hours we will return the cached prisons. Once the TTL has expired redis will evict. At 24 hours and 1 second, we will not re populate the cache. We will have an empty cache until the next time someone requests the prisons, and which point we will get a fresh set of prisons from the API and cache them for a new 24 hour period.
Make sense ?